src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,367 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Kiota.Builder.CodeDOM;
+using Kiota.Builder.Extensions;
+using Microsoft.Kiota.Abstractions;
+
+namespace Kiota.Builder.Writers.Http;
+public class CodeClassDeclarationWriter(HttpConventionService conventionService) : CodeProprietableBlockDeclarationWriter<ClassDeclaration>(conventionService)
+{
+    private const string BaseUrlPropertyName = "hostAddress";
+
+    protected override void WriteTypeDeclaration(ClassDeclaration codeElement, LanguageWriter writer)
+    {
+        ArgumentNullException.ThrowIfNull(codeElement);
+        ArgumentNullException.ThrowIfNull(writer);
+
+        if (codeElement.Parent is CodeClass requestBuilderClass && requestBuilderClass.IsOfKind(CodeClassKind.RequestBuilder) && GetUrlTemplateProperty(requestBuilderClass) is CodeProperty urlTemplateProperty)
+        {
+            // Write short description
+            conventions.WriteShortDescription(requestBuilderClass, writer);
+            writer.WriteLine();
+
+            // Retrieve all query parameters
+            var queryParameters = GetAllQueryParameters(requestBuilderClass);
+
+            // Retrieve all path parameters
+            var pathParameters = GetPathParameters(requestBuilderClass);
+
+            var baseUrl = GetBaseUrl(requestBuilderClass);
+
+            // Write path parameters
+            WritePathParameters(pathParameters, writer);
+
+            // Write all query parameter variables
+            WriteQueryParameters(queryParameters, writer);
+
+            // Write all HTTP methods GET, POST, PUT, DELETE e.t.c
+            WriteHttpMethods(requestBuilderClass, writer, queryParameters, pathParameters, urlTemplateProperty, baseUrl);
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all the query parameters for the given request builder class.
+    /// </summary>
+    /// <param name="requestBuilderClass">The request builder class containing the query parameters.</param>
+    /// <returns>A list of all query parameters.</returns>
+    private static List<CodeProperty> GetAllQueryParameters(CodeClass requestBuilderClass)
+    {
+        var queryParameters = new List<CodeProperty>();
+
+        // Retrieve all the query parameter classes
+        var queryParameterClasses = requestBuilderClass
+            .GetChildElements(true)
+            .OfType<CodeClass>()
+            .Where(static element => element.IsOfKind(CodeClassKind.QueryParameters))
+            .ToList();
+
+        // Collect all query parameter properties into the aggregated list
+        queryParameterClasses?.ForEach(paramCodeClass =>
+        {
+            var queryParams = paramCodeClass
+                .Properties
+                .Where(static property => property.IsOfKind(CodePropertyKind.QueryParameter))
+                .ToList();
+
+            queryParameters.AddRange(queryParams);
```

Comment:
use a select many instead, and combine the linq statements to reduce allocations